tools/restatectl/src/commands/partition/list.rs

```diff
@@ -79,27 +107,85 @@ pub async fn list_partitions(
                     let host_node =
                         GenerationalNodeId::new(host.id, host.generation.expect("generation"));
                     let details = PartitionListEntry { host_node, status };
-                    partitions.entry(partition_id).or_default().push(details);
+                    partitions.push((partition_id, details));
                 }
             }
         }
     }
+
     // Show information organized by partition and node
     let mut partitions_table = Table::new_styled();
     partitions_table.set_styled_header(vec![
         "P-ID",
         "NODE",
         "MODE",
         "STATUS",
-        "APPLIED",
-        "PERSISTED",
         "LEADER",
         "EPOCH",
+        "SEQUENCER",
+        "APPLIED",
+        "PERSISTED",
         "SKIPPED",
         "LAST-UPDATE",
     ]);
-    for (partition_id, processors) in partitions {
-        for processor in processors {
+
+    partitions
+        .into_iter()
+        .sorted_by(|a, b| match opts.sort {
+            SortMode::Partition => a.0.cmp(&b.0),
+            SortMode::Node => {
+                a.1.host_node
+                    .cmp(&b.1.host_node)
+                    .then_with(|| a.0.cmp(&b.0))
+            }
+            SortMode::Active => {
+                a.1.status
+                    .effective_mode
+                    .cmp(&b.1.status.effective_mode)
+                    .then_with(|| a.1.host_node.cmp(&b.1.host_node))
+            }
+        })
+        .for_each(|(partition_id, processor)| {
+            let is_leader = processor
+                .status
+                .last_observed_leader_node
+                .map(|n| {
+                    n.generation.is_some_and(|g| {
+                        PlainNodeId::from(n.id).with_generation(g) == processor.host_node
+                    })
+                })
+                .unwrap_or_default();
```

Comment:
Does this mean that there can be two leaders in the output if there are two pps which have observed different leader nodes (one still seeing the old one and the other observing the newer one)?

Reply:
Maybe we use one way to distinguish leaders from non-leaders. Here we make the decision based on the last observed_leader_node. In the `render_mode` we base it on the planned/effective mode.

Reply:
> Does this mean that there can be two leaders in the output if there are two pps which have observed different leader nodes (one still seeing the old one and the other observing the newer one)?

Yes, this is absolutely true! The tool reports what the admin node sees. This is what I'm trying to draw attention to with the highlighting logic. A further improvement is to inspect the PP epoch, which will tell us if some of these PPs have outdated views. Here, the `is_leader` flag is really `pp_sees_itself_as_leader` - I've renamed it to convey this. But the flag is only used to highlight the displayed observed leader from this node, nothing else.

The `render_mode` is more about the transitional status particularly when the two are different; I've now also added color highlighting when a PP's observed leadership epoch is not the latest.